py/picca/delta_extraction/data.py

```diff
@@ -150,9 +148,11 @@ def __parse_config(self, config):
                 pixel_step_rest_frame = pixel_step
                 self.logger.info("'delta lambda rest frame' not set, using "
                                  f"the same value as for 'delta lambda' ({pixel_step_rest_frame})")
-        else:
-            raise DataError("Forest.wave_solution must be either "
-                            "'log' or 'lin'")
+        # this should not be reached as wave_solution is either "lin" or "log"
+        # added here only in case we add another wave_solution in the future
```

Comment:
No, the value of `wave_solution` here is the one used to initialize the `Forest` class variable. Initially, I thought to remove it (as you can see in the commit you're signaling) since I did not know how to have `coverall` ignore certain branches. Then I figured it would be better to have this check to mark this line if we decide to include something else in the future. Otherwise, we risk changing the `assert` above (line 127) and unknowingly add a bug.

Reply:
I can imagine a different mode if e.g., at some point, we decide to directly load data from multiple surveys